test/functional/testlib.bash

```diff
@@ -1188,6 +1201,42 @@ update_manifest() { # swupd_function
 
 }
 
+get_entry_from_manifest() { # swupd_function
+
+	local manifest=$1
+	local entry_key=$2
+	local value
+
+	# If no parameters are received show usage
+	if [ $# -eq 0 ]; then
+		cat <<-EOM
+			Usage:
+			    get_entry_from_manifest <manifest> [file_hash or file_name]
+			EOM
+		return
+	fi
+	validate_item "$manifest"
+
+	# if no key is provided, return a list with all
+	# files/entries from the manifest
+	if [ -z "$entry_key" ]; then
+		sudo cat "$manifest" | grep -E -x "(D|F|L)..."$'\t'".*"$'\t'".."$'\t'"/.*" | awk '{ print $4 }'
```

Comment:
For me, this doesn't work when version number of file has more than 2 digits.

Reply:
ahh, good catch my friend, I'll fix that